central/scannerdefinitions/handler/handler.go

```diff
@@ -341,34 +330,38 @@ func (h *httpHandler) openOfflineDefinitions(ctx context.Context, t updaterType,
 		// We want to find a specific file inside the toplevel online file.
 		fallthrough
 	case mappingUpdaterType:
-		// search mapping file
+		// Search mapping file.
 		fileName := filepath.Base(opts.fileName)
 		targetFile, _, err := h.openFromArchive(offlineBlob.Name(), fileName)
 		if err != nil {
 			return nil, err
 		}
 		offlineFile = &vulDefFile{File: targetFile, modTime: offlineBlob.modTime}
 	case vulnerabilityUpdaterType:
-		// check version information in manifest
+		// Check version information in manifest.
 		mf, _, err := h.openFromArchive(offlineBlob.Name(), scannerV4ManifestFile)
 		if err != nil {
+			log.Errorf("open manifest: %v", err)
 			return nil, err
 		}
 		defer utils.IgnoreError(mf.Close)
 
-		v, err := readV4ManifestVersion(mf)
+		mv, err := readV4Manifest(mf)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("reading v4 definitions manifest: %w", err)
 		}
 
-		if v != minorVersionPattern.FindString(opts.vulnVersion) && (opts.vulnVersion != "dev" || buildinfo.ReleaseBuild) {
-			msg := fmt.Sprintf("failed to get Scanner V4 offline data file: the offline data version != requested file version (%s != %s)", v, opts.vulnVersion)
-			log.Error(msg)
-			return nil, errors.New(msg)
+		// Only validate offline definition versions on release builds.
+		if buildinfo.ReleaseBuild && mv.VulnerabilityVersion != opts.vulnVersion {
+			log.Debugf("v4 offline bundle vulnerability version mismatch: "+
+				"the stored version != requested vulnerability version (%s != %s)",
+				mv.VulnerabilityVersion, opts.vulnVersion)
+			return nil, nil
```

Comment:
curious why change this to a `nil` error? What will be output in the logs/error returned to Scanner V4?